src/Connect/connect.cpp

```diff
@@ -39,6 +40,30 @@ namespace con {
 
 namespace {
 
+    std::atomic<OnlineStatus> last_known_status = OnlineStatus::Unknown;
+
+    optional<OnlineStatus> err_to_status(const Error error) {
```

Comment:
why does it have `optional `return value and return `OnlineStatus::Unknown;`

Reply:
You're correct. That must have been some leftover from figuring out the exact stuff as I went 😇. Fixed.